browser-test/src/applicant/application_version_fast_forward.test.ts

```diff
@@ -0,0 +1,1000 @@
+   * @constructor
+   * @param {string} programName name of the program
+   * @param {Page} page Unique page instance for this actor
+   */
+  private constructor(programName: string, page: Page) {
+    this.programName = programName
+    this.page = page
+    this.adminPrograms = new AdminPrograms(page)
+  }
+
+  /**
+   * Simplifies creation of the {FastForwardProgramAdminActor}
+   * @param {string} programName
+   * @param {Browser} browser instance from Playwright
+   * @returns {Promise<FastForwardProgramAdminActor>} Promise to an instance of this class
+   */
+  public static async create(
+    programName: string,
+    browser: Browser,
+  ): Promise<FastForwardProgramAdminActor> {
+    const context = await browser.newContext()
+    return new FastForwardProgramAdminActor(
+      programName,
+      await context.newPage(),
+    )
+  }
+
+  /**
+   * Log in the program admin actor
+   */
+  async login() {
+    await this.page.goto('/programs')
+    await closeWarningMessage(this.page)
+    await loginAsProgramAdmin(this.page)
+  }
+
+  /**
+   * Log out the program admin actor
+   */
+  async logout() {
+    await logout(this.page)
+  }
+
+  /**
+   * Navigate to the view applications page at `/admin/programs/:programId/applications
+   */
+  async viewApplications() {
+    await this.adminPrograms.viewApplications(this.programName)
+  }
+
+  /**
+   * Get a locator any admin application cards. There may be 0-♾️
+   * @returns {Locator} Locator to admin application card
+   */
+  getCardLocator(): Locator {
+    return this.page.locator('.cf-admin-application-card')
+  }
+
+  /**
+   * Gets the program id from the href of the locator
+   * @param {Locator} locator to an admin application card element containing an href to the application
+   * @returns {number} programId. If the ID is not found a failed assertion will end the test.
+   */
+  async parseProgramIdFromLocator(locator: Locator) {
+    const valueToSearch = (await locator.getAttribute('href')) || ''
+    const pattern = /\/admin\/programs\/(?<id>[0-9]*)\/applications\/*/
+
+    const id = parseInt(valueToSearch?.match(pattern)?.groups?.id || '-1')
+    expect(id).toBeGreaterThan(0)
+
+    return id
+  }
+}
+
+/**
+ * Test suite infernal definition of a block and its questions configuration
+ */
+interface BlockDefinition {
+  block: Block
+  questions: Question[]
+  eligibilityValue?: string
+}
+
+/**
+ * List of all question names used in this test suite
+ */
+enum Question {
+  A = 'question-name-a',
+  B = 'question-name-b',
+  C = 'question-name-c',
+  D = 'question-name-d',
+  E = 'question-name-e',
+  F = 'question-name-f',
+  G = 'question-name-g',
+  H = 'question-name-h',
+}
+
+/**
+ * List of all block names used in this test suite
+ */
+enum Block {
+  First = 'Screen 1',
+  Second = 'Screen 2',
+  Third = 'Screen 3',
+  Fourth = 'Screen 4',
+  Fifth = 'Screen 5',
+}
+
+/**
+ * List of application statuses used in this test suite
+ */
+enum ApplicationStatus {
```

Comment:
**thought**: it might be nice to call this something else to disambiguate from the actual ApplicationStatus field on applications.

idk what else though. `UserFacingApplicationStatus` is a bit clunky.

Reply:
Yeah. It's not really a _common_ term in the application beyond the UI and is mainly geared to what is on `/programs`. Renaming it to `ApplicationStatusCardGroupName`. While just calling it ApplictionCareGroupName might be more accurate I feel including _status_ in it adds some clarity in the test.